src/services/organizeImports.ts

```diff
@@ -0,0 +1,291 @@
+/* @internal */
+namespace ts.OrganizeImports {
+    export function organizeImports(
+        sourceFile: SourceFile,
+        formatContext: formatting.FormatContext,
+        host: LanguageServiceHost,
+        cancellationToken: CancellationToken) {
+
+        // All of the (old) ImportDeclarations in the file, in syntactic order.
+        const oldImportDecls: ImportDeclaration[] = [];
+
+        forEachChild(sourceFile, node => {
+            cancellationToken.throwIfCancellationRequested();
+            if (isImportDeclaration(node)) {
+                oldImportDecls.push(node);
+            }
+            // TODO (https://github.com/Microsoft/TypeScript/issues/10020): sort *within* ambient modules (find using isAmbientModule)
+        });
+
+        if (oldImportDecls.length === 0) {
+            return [];
+        }
+
+        const usedImportDecls = removeUnusedImports(oldImportDecls);
+        cancellationToken.throwIfCancellationRequested();
+        const sortedImportDecls = sortImports(usedImportDecls);
+        cancellationToken.throwIfCancellationRequested();
+        const coalescedImportDecls = coalesceImports(sortedImportDecls);
+        cancellationToken.throwIfCancellationRequested();
+
+        // All of the (new) ImportDeclarations in the file, in sorted order.
+        const newImportDecls = coalescedImportDecls;
+
+        const changeTracker = textChanges.ChangeTracker.fromContext({ host, formatContext });
+
+        // NB: Stopping before i === 0
+        for (let i = oldImportDecls.length - 1; i > 0; i--) {

```

Comment:
Could you add a comment explaining why this is in reverse order? I assume this has something to do with trivia?

Reply:
It's not - I was thinking of something else.